src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let result = sourceString.split(';');
+
+  result = result.filter((item) => item !== '');
+  result = result.map((item) => item.split(':').map((elem) => elem.trim()));
+
+  result = result.reduce((acc, item) => {
+    acc[item[0]] = item[1];
+
+    return acc;
```

Comment:
The reduce function can be chained directly after the map function without reassigning 'result' each time. This will make the code cleaner and more readable. Also, ensure proper indentation when chaining methods.